pkg/clients/edgeconnect/iface.go

```diff
@@ -6,10 +6,10 @@ type Client interface {
 	GetEdgeConnect(edgeConnectId string) (GetResponse, error)
 
 	// CreateEdgeConnect creates edge connect
-	CreateEdgeConnect(name string, hostPatterns []string, oauthClientId string) (CreateResponse, error)
+	CreateEdgeConnect(name string, hostPatterns []string, k8sAutomationHostPattern string, oauthClientId string) (CreateResponse, error)
```

Comment:
Create host pattern mapping on this level and pass it into `CreateEdgeConnect`, so it's more flexible should we need to add another host pattern mapping in the future.
```suggestion
        hostPatternMappings := map[string]string { k8sAutomationHostPattern: defaultKubnertesApiDns }
	CreateEdgeConnect(name string, hostPatterns []string,
        hostPatternMappings, oauthClientId string) (CreateResponse, error)
```

Reply:
That's a great idea actually :+1: OMW

Reply:
Fixed [here](https://github.com/Dynatrace/dynatrace-operator/pull/3332/commits/11075d6cd8e6e211a774059eae3c459c68864c68)
I refactored quite a bit, but I think it is better this way. Now we have an edgeConnect.HostMappings() function the same way as we have the HostPatterns() one.